### PR TITLE
Fix quoting for command line args

### PR DESCRIPTION
**Pull request by uasi**
# Description

Fixes https://github.com/nushell/nushell/issues/5370
Fixes https://github.com/nushell/nushell/issues/5371

# Tests

Make sure you've run and fixed any issues with these commands:

- [x] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [x] `cargo clippy --all --all-features -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [x] `cargo build; cargo test --all --all-features` to check that all the tests pass

### CODE REVIEW

**Comment by fdncred** [comment]
seems a reasonable approach. let's see how it works. Thanks!

File: src/main.rs [hunk 1]
```diff
@@ -79,36 +79,28 @@ fn main() -> Result<()> {
...
-                || arg == "--threads"
-                || arg == "-t"
-            {
-                collect_arg_nushell = true;
-            }
+            let flag_value = match arg.as_ref() {
+                "--commands" | "-c" => {
+                    // FIXME: Use proper quoting. `escape_quote_string()` can't be used for now due to https://github.com/nushell/nushell/issues/5383.
+
+                    args.next().map(|a| format!("`{}`", a))
```
Comment (uasi) [inline src/main.rs:92]: Using proper quoting will fix https://github.com/nushell/nushell/issues/5175

Comment (fdncred) [reply src/main.rs:92]: Feel free to propose a 'proper quoting' fix too. :)

**Comment by elferherrera** [comment]
I think this PR caused this issue
image

**Comment by jntrnr** [comment]
@elferherrera I just landed https://github.com/nushell/nushell/pull/5398. Can you see if it fixes your issue?

**Comment by elferherrera** [comment]
it seems that it didnt work
image

**Comment by jntrnr** [comment]
@elferherrera this should now be fixed in the latest main. Took a few tries, but I think I found it

**Pull request merged**